modules/core/05-port/types/ibc_legacy_module.go

```diff
@@ -74,6 +74,50 @@ func (im *LegacyIBCModule) OnChanOpenInit(
 	return reconstructVersion(im.cbs, negotiatedVersions)
 }
 
+// OnChanOpenTry implements the IBCModule interface.
+func (im *LegacyIBCModule) OnChanOpenTry(
```

Comment:
this is almost an exact duplicate, I played around with trying to generalize the Init/Try functions, but it ended up looking even uglier, happy to revisit this idea, but I think it's fine as is for now unless someone has a suggestion for removing the duplication.